hack/update/kubeadm_constants/update_kubeadm_constants.go

```diff
@@ -0,0 +1,205 @@
+			data = Data{ImageMap: imageMapString}
+			schema[minikubeConstantsFilePath].Replace[`KubeadmImages = .*`] =
+				`KubeadmImages = map[string]map[string]string{ {{.ImageMap}}`
+		} else {
+			data = Data{ImageMap: strings.TrimLeft(imageMapString, "\n")}
+			versionIdentifier := fmt.Sprintf(`"%s": {[^}]+},`, majorMinorVersion)
+			schema[minikubeConstantsFilePath].Replace[versionIdentifier] = "{{.ImageMap}}"
+		}
+
+		update.Apply(ctx, schema, data, "", "", -1)
+	}
+}
+
+func getKubeadmImagesMapString(version string) (string, error) {
+	url := fmt.Sprintf(kubeadmReleaseURL, version)
+	fileName := fmt.Sprintf(kubeadmBinaryName, version)
+	if err := downloadFile(url, fileName); err != nil {
+		klog.Errorf("failed to download kubeadm binary %s", err.Error())
+		return "", err
+	}
+
+	kubeadmCommand := fmt.Sprintf("./%s", fileName)
+	args := []string{"config", "images", "list"}
+	imageListString, err := executeCommand(kubeadmCommand, args...)
+	if err != nil {
+		klog.Errorf("failed to execute kubeadm command %s", kubeadmCommand)
+		return "", err
+	}
+
+	if err := os.Remove(fileName); err != nil {
+		klog.Errorf("failed to remove binary %s", fileName)
+	}
+
+	return formatKubeadmImageList(version, imageListString)
+}
+
+func formatKubeadmImageList(version, data string) (string, error) {
+	templateData := make(map[string]map[string]string)
+	majorMinorVersion := semver.MajorMinor(version)
+	templateData[majorMinorVersion] = make(map[string]string)
+	lines := strings.Split(data, "\n")
+	for _, line := range lines {
+		imageTag := strings.Split(line, ":")
+		if len(imageTag) == 2 {
+			templateData[majorMinorVersion][imageTag[0]] = imageTag[1]
+		}
+	}
+
+	imageTemplate := template.New("kubeadmImage")
+	t, err := imageTemplate.Parse(kubeadmImagesTemplate)
+	if err != nil {
+		klog.Errorf("failed to create kubeadm image map template %s", err.Error())
+		return "", err
+	}
+
+	var bytesBuffer bytes.Buffer
+	if err := t.Execute(&bytesBuffer, &templateData); err != nil {
+		return "", err
+	}
+
+	return bytesBuffer.String(), nil
+}
+
+func downloadFile(url, fileName string) error {
+	file, err := os.Create(fileName)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	response, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+	defer response.Body.Close()
+
+	if response.StatusCode != http.StatusOK {
+		return fmt.Errorf("non success status code, while downloading file: %s from: %s", fileName, url)
+	}
+
+	if _, err := io.Copy(file, response.Body); err != nil {
+		return err
+	}
+
+	return os.Chmod(fileName, os.ModePerm)
+}
+
+func executeCommand(command string, args ...string) (string, error) {
+	output, err := exec.Command(command, args...).Output()
+	if err != nil {
+		return "", err
+	}
+	return string(output), nil
+}
+
+// getK8sVersion returns Kubernetes versions.
+func getK8sVersions(ctx context.Context, owner, repo string) (stable, latest, latestMM, latestP0 string, err error) {
```

Comment:
`latestMM` & `latestP0` aren't used anywhere, couldn't we just replace this whole function with a direct call to `update.GHReleases`?